static/src/javascripts/bootstraps/enhanced/newsletters.js

```diff
@@ -114,6 +123,7 @@ const showSecondStageSignup = (buttonEl: HTMLButtonElement): void => {
     fastdom.write(() => {
         buttonEl.setAttribute('type', 'button');
         bean.on(buttonEl, 'click', () => {
+            modifyAtSecondStage(buttonEl)
```

Comment:
I think modifying the link name is part of "showing the signup form" so I'd probably put the functionality in there (even if you keep the separate functions to actually do the modifications).